test/Microsoft.AspNetCore.Razor.Test.Common/Language/IntegrationTests/IntegrationTestBase.cs

```diff
@@ -220,9 +220,102 @@ protected void AssertSourceMappingsMatchBaseline(RazorCodeDocument document)
             var baseline = testFile.ReadAllText();
 
             // Normalize newlines to match those in the baseline.
-            var actual = serializedMappings.Replace("\r", "").Replace("\n", "\r\n");
+            var actualBaseline = serializedMappings.Replace("\r", "").Replace("\n", "\r\n");
 
-            Assert.Equal(baseline, actual);
+            Assert.Equal(baseline, actualBaseline);
+
+            var syntaxTree = codeDocument.GetSyntaxTree();
+            var visitor = new CodeSpanVisitor();
+            visitor.VisitBlock(syntaxTree.Root);
+
+            var charBuffer = new char[codeDocument.Source.Length];
+            codeDocument.Source.CopyTo(0, charBuffer, 0, codeDocument.Source.Length);
+            var sourceContent = new string(charBuffer);
+
+            var spans = visitor.CodeSpans;
+            for (var i= 0; i < spans.Count; i++)
+            {
+                var span = spans[i];
+                if (span.Start.FilePath == null || span.Start.FilePath != codeDocument.Source.FilePath)
+                {
+                    // Not in the main file, skip.
+                    continue;
+                }
+
+                var location = new SourceSpan(span.Start, span.Length);
+                var expectedSpan = sourceContent.Substring(span.Start.AbsoluteIndex, span.Length);
+
+                // See #
```

Comment:
I'm planning to log a bug for this, which we will likely end up closing. 

We can fix the issue in the parser but it impacts the completion experience. For right now we're better off with the bug since the workaround in the verifier is localized.